### PR TITLE
Chrome full size screenshot without sticky header

### PR DESCRIPTION
**Pull request by galkina-anna**

### CODE REVIEW

File: src/main/java/com/assertthat/selenium_shutterbug/utils/web/DriverCommandExecutor.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.openqa.selenium.remote.http.HttpMethod;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+/**
+ * Created by Anna Galkina on 23-06-2018.
+ */
+public class DriverCommandExecutor {
```
Comment (glib-briia) [inline src/main/java/com/assertthat/selenium_shutterbug/utils/web/DriverCommandExecutor.java]: Lets move command execution methods to Browser class

File: src/main/resources/js/all-metrics.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+({width: Math.max(window.innerWidth,document.body.scrollWidth,document.documentElement.scrollWidth)|0,
+height: Math.max(window.innerHeight,document.body.scrollHeight,document.documentElement.scrollHeight)|0,
+deviceScaleFactor: window.devicePixelRatio || 1,mobile: typeof window.orientation !== 'undefined'})
```
Comment (glib-briia) [inline src/main/resources/js/all-metrics.js]: Add newline at the end of file

File: src/main/java/com/assertthat/selenium_shutterbug/core/Shutterbug.java [hunk 3]
```diff
@@ -97,7 +97,7 @@ public static PageSnapshot shootPage(WebDriver driver, ScrollStrategy scroll, in
             case BOTH_DIRECTIONS:
                 pageScreenshot.setImage(browser.takeScreenshotEntirePage());
             case WITHOUT_STICKY_HEADER:
```
Comment (glib-briia) [inline src/main/java/com/assertthat/selenium_shutterbug/core/Shutterbug.java]: Lets call it WHOLE_PAGE_CHROME

File: src/main/java/com/assertthat/selenium_shutterbug/core/Shutterbug.java [hunk 4]
```diff
@@ -97,7 +97,7 @@ public static PageSnapshot shootPage(WebDriver driver, ScrollStrategy scroll, in
...
-                pageScreenshot.setImage(browser.takeScreenshotEntirePage(true));
+                pageScreenshot.setImage(browser.takeScreenshotEntirePageUsingChromeCommand());
```
Comment (glib-briia) [inline src/main/java/com/assertthat/selenium_shutterbug/core/Shutterbug.java]: Shall we add broser.getName(); which will check which driver instance is currently used (instanceof presumably) and throw UnsupportedOperationException if current driver instance is other than Chrome

Comment (glib-briia) [reply src/main/java/com/assertthat/selenium_shutterbug/core/Shutterbug.java]: Also, please add missing break to BOTH_DIRECTIONS case

File: src/main/java/com/assertthat/selenium_shutterbug/utils/web/Browser.java [hunk 5]
```diff
@@ -111,6 +115,33 @@ public BufferedImage takeScreenshotEntirePage() {
...
+        }
+        int verticalIterations = (int) Math.ceil(((double) this.getDocHeight()) / this.getViewportHeight());
+        for (int j = 0; j < verticalIterations; j++) {
+            this.scrollTo(0, j * this.getViewportHeight());
+            wait(scrollTimeout);
+        }
+        Object metrics = driver.evaluate(FileUtil.getJsScript(ALL_METRICS));
+        driver.sendCommand("Emulation.setDeviceMetricsOverride", metrics);
+        Object result = driver.sendCommand("Page.captureScreenshot", ImmutableMap.of("format", "png", "fromSurface", true));
+        driver.sendCommand("Emulation.clearDeviceMetricsOverride", ImmutableMap.of());
```
Comment (glib-briia) [inline src/main/java/com/assertthat/selenium_shutterbug/utils/web/Browser.java]: It doesn't look like we need to clearDeviceMetricsOverride, as we don't really changing any initial device metrics just width and height which wont influence further test flow, moreover clearDeviceMetricsOverride for some reason spoins further highlighting/bluring etc. elements (coordinates are shifted).

**Comment by glib-briia** [comment]
Update from master before resolving.

**Comment by galkina-anna** [comment]
Hi @glib-briia !

Thank you a lot for your review! 

I have fixed my code following your comments. Only one left  - I didn't change the doc as on README file this information would be inappropriate and wiki I have no access to amend.

Could you please do this for me or tell me how I find the way to change the doc?

Thank you,
Anna

File: src/main/resources/js/all-metrics.js (hunk 2 above)
Comment (galkina-anna) [reply src/main/resources/js/all-metrics.js]: New line added, but here in PR it is not showed:
image

**Pull request merged**